x/concentrated-liquidity/pool.go

```diff
@@ -139,15 +139,17 @@ func (k Keeper) CalcOutAmtGivenIn(ctx sdk.Context,
 		liquidity:                p.Liquidity,
 	}
 
-	for swapState.amountSpecifiedRemaining.GT(sdk.ZeroDec()) && !swapState.sqrtPrice.Equal(sqrtPriceLimit) {
+	// had to reintroduce the dust check, we need to figure out a cleaner way to do this
+	for swapState.amountSpecifiedRemaining.GT(sdk.NewDecWithPrec(1, 8)) && !swapState.sqrtPrice.Equal(sqrtPriceLimit) {
```

Comment:
So I personally think we should not reintroduce dust check.
I was digging around with this for loop yesterday, and the closest answer I got to was that it only works with the dust check not because of some sort of minor rounding issues, its because that the calculation being done is off(incorrect) and we're just force exiting from the loop. 

I, myself have to do more investigation on this, and I'm also fine with leaving the dust check for now, but just wanted to let you know that having the dust check means that we still have a point that we missed somewhere!

Reply:
Agreed this is not something we want long term, but my vs code was having a seizure every time I accidentally reintroduced an inf loop so wanted something to prevent this. Maybe we should introduce a max 100 loop counter for now, wdyt?